File: TwoPointer/ThreeSumClosest.java
import java.util.Arrays;

public class ThreeSumClosest {
    public static void main(String[] args) {
        ThreeSumClosest solution = new ThreeSumClosest();
        int[] nums = { -1, 2, 1, 4 };
        int target = 1;
        int ans = solution.input(nums, target);
        System.out.println(ans);

    }

    public int input(int[] nums, int target) {
        Arrays.sort(nums);
        int ans = Integer.MAX_VALUE;
        for (int i = 0; i < nums.length; i++) {
            int l = i + 1;
            int r = nums.length - 1;

            while (l < r) {
                int sum = nums[i] + nums[l] + nums[r];
                if (Math.abs(target - sum) < Math.abs(ans))
                    ans = target - sum;

                if (sum > target)
                    r--;
                else
                    l++;
            }
        }
        return (target - ans);
    }
}

File: TwoPointer/ThreeSumClosest.js
var threeSumClosest = function(nums, target) {
   nums.sort((a,b)=>a-b);
   let closesum=Infinity;
   let min=Infinity;
   for(let i=0;i<nums.length;i++){
    let left=i+1;
    let right=nums.length-1;
    while(left<right){
        let sum=nums[i]+nums[left]+nums[right];
        let diff=Math.abs(sum-target);
        if(diff<min){
            min=diff;
            closesum=sum;
        }
        if(sum<target){
            left++;
        }else{
            right--;
        }
    }
   }
   return closesum;
};
const nums=[-1,2,1,-4];
const target=1;
const result=threeSumClosest(nums,target);
console.log(result);